src/builder.rs

```diff
@@ -251,21 +251,37 @@ impl<'req, 'win, W: HasRawWindowHandle> PixelsBuilder<'req, 'win, W> {
             swap_chain,
             texture,
             texture_extent,
+            texture_format: self.texture_format,
             texture_format_size,
             scaling_renderer,
         };
-
-        Ok(Pixels {
+        let mut pixels = Pixels {
             context,
             surface_size,
             present_mode,
             pixels,
             scaling_matrix_inverse,
             render_texture_format: self.render_texture_format,
-        })
+        };
+        create_swap_chain(&mut pixels);
+
+        Ok(pixels)
     }
 }
 
+pub(crate) fn create_swap_chain(pixels: &mut Pixels) {
```

Comment:
Gut feeling is that I would make this a method of the `Pixels` struct instead of a module-level function on `builder`. Thinking I see why you did it this way, though.

There's a call to `create_swap_chain` further up (line 222) that could be removed _iff_ this was a static method that accepted free arguments instead of an instance method... Kind of like it is right now. And that would be fine.

Other options are changing the type of `Context::swap_chain` to `Option<T>` and then unwrapping it everywhere, or trying to create a half-initialized structure using `unsafe` and [`MaybeUninit`](https://doc.rust-lang.org/nightly/std/mem/union.MaybeUninit.html#initializing-a-struct-field-by-field). I'm not a big fan of `unsafe` though. And having to `unwrap()` the swap chain to use it seems unfortunate. Given that, I think I prefer the static method with free args.

Reply:
I'm missed the useless call at line 222 (the downsize of coding at 4 AM :) ). I'm looking at it.

Reply:
The useless call to `device.create_swap_chain` line 222 has been removed.
Now we have:

- in builder.rs
/// Generic swap chain creation with free arguments
`pub(crate) fn create_swap_chain(device: &mut wgpu::Device, surface: &wgpu::Surface, format: wgpu::TextureFormat,
    width: u32, height: u32, present_mode: wgpu::PresentMode)`

- in lib.rs
/// Method of Pixels, which calls above function with own values
`pub(crate) fn re_create_swap_chain(&mut self) { /* calls create_swap_chain above, with own values */ }`